polkadot/runtime/common/src/impls.rs

```diff
@@ -67,6 +67,52 @@ where
 	}
 }
 
+pub struct EraPayoutParams {
+	pub total_staked: Balance,
+	pub total_stakable: Balance,
+	pub ideal_stake: Perquintill,
+	pub max_annual_inflation: Perquintill,
+	pub min_annual_inflation: Perquintill,
+	pub falloff: Perquintill,
+	pub period_fraction: Perquintill,
+	pub legacy_auction_proportion: Option<Perquintill>,
+}
+
+pub fn relay_era_payout(params: EraPayoutParams) -> (Balance, Balance) {
+	let EraPayoutParams {
+		total_staked,
+		total_stakable,
+		ideal_stake,
+		max_annual_inflation,
+		min_annual_inflation,
+		falloff,
+		period_fraction,
+		legacy_auction_proportion,
+	} = params;
+
+	let delta_annual_inflation = max_annual_inflation.saturating_sub(min_annual_inflation);
+
+	let ideal_stake = ideal_stake.saturating_sub(legacy_auction_proportion.unwrap_or_default());
+
+	let stake = Perquintill::from_rational(total_staked, total_stakable);
+	let adjustment = pallet_staking_reward_fn::compute_inflation(stake, ideal_stake, falloff);
+	let staking_inflation =
+		min_annual_inflation.saturating_add(delta_annual_inflation * adjustment);
+
+	let max_payout = period_fraction * max_annual_inflation * total_stakable;
+	let staking_payout = (period_fraction * staking_inflation) * total_stakable;
+	let rest = max_payout.saturating_sub(staking_payout);
+
+	let other_issuance = total_stakable.saturating_sub(total_staked);
+	if total_staked > other_issuance {
+		let _cap_rest = Perquintill::from_rational(other_issuance, total_staked) * staking_payout;
+		// We don't do anything with this, but if we wanted to, we could introduce a cap on the
+		// treasury amount with: `rest = rest.min(cap_rest);`
+	}
+	(staking_payout, rest)
+}
+
+#[deprecated = "use relay_era_payout instead which supports dynamic parameters"]
```

Comment:
The main difference between deprecated `era_payout` and the new one is cleaner interface, and the implied deprecation of auction. The interface of `relay_era_payout` does not actually say anything about dynamic parameters, right? 

this is a good example of leaking implementation detail (how _we intend to use this_) into the interface (how _someone else might use this_). Best not to mix the two :)

Reply:
```suggestion
#[deprecated = "use relay_era_payout instead for an improved interface without auction functionality"]
```
I'm not sure if this wording is the best

Reply:
Why do we need to keep the old function? This is the westend runtime, nobody besides the runtime calls these functions. I think you can just move it into the test module?

Reply:
Hmm doesn't the fellowship runtimes use it? If not, then yes, I am happy to remove it.

Reply:
Let's only keep it for legacy testing.

Reply:
Moved the fn into the test module